modules/openapi-generator/src/main/resources/python/configuration.mustache

```diff
@@ -349,41 +349,37 @@ class Configuration(object):
             {{/servers}}
         ]
 
-    def get_host_from_settings(self, index, variables={}):
-        """Gets host URL based on the index and variables
+    def get_host_from_settings(self, index, variables=None):
+        """Get host URL based on the index and variables.
+
         :param index: array index of the host settings
-        :param variables: hash of variable and the corresponding value
+        :param variables: mapping of variable and the corresponding value
         :return: URL based on host settings
         """
-
+        variables = {} if variables is None else variables
         servers = self.get_host_settings()
 
-        # check array index out of bound
-        if index < 0 or index >= len(servers):
+        try:
+            server = servers[index]
+        except IndexError:
             raise ValueError(
-                "Invalid index {} when selecting the host settings. Must be less than {}"  # noqa: E501
+                "Invalid index {0} when selecting the host settings. Must be less than {1}"  # noqa: E501
                 .format(index, len(servers)))
 
-        server = servers[index]
         url = server['url']
 
         # go through variable and assign a value
         for variable_name in server['variables']:
```

Comment:
Hey, you are iterating through a dict and using its key and value here.
Why not use `for var_name, var_value_dict in six.iteritems(server['variables']):`?